test/controllers/integrations/buildkite_controller_test.rb

```diff
@@ -63,4 +63,29 @@
       response.status.must_equal 200
     end
   end
+
+  context 'when the buildkite_release_params hook gets trigger' do
+    before do
+      project.releases.destroy_all
+      project.builds.destroy_all
+      Integrations::BuildkiteController.any_instance.stubs(:deploy?).returns(true)
+      Integrations::BuildkiteController.any_instance.stubs(:project).returns(project)
+      Integrations::BuildkiteController.any_instance.stubs(:commit).returns(commit)
+      Integrations::BuildkiteController.any_instance.stubs(:branch).returns('master')
+      Project.any_instance.stubs(:create_releases_for_branch?).returns(true)
+      Build.any_instance.stubs(:validate_git_reference).returns(true)
+      stub_request(:post, "https://api.github.com/repos/bar/foo/releases").
+        to_return(status: 200, body: "", headers: {})
+      Samson::Hooks.callback :buildkite_release_params do |_|
```

Comment:
We've noticed that there is a Samson::Hooks.with_callback method which will do the clean up after testing is finished, so we are using it now, please see the latest commit for more info